lib/sensors/accelerometer.cpp

```diff
@@ -0,0 +1,137 @@
+#include "accelerometer.hpp"
+
+namespace hyped::sensors {
+
+Accelerometer::Accelerometer(core::ILogger &logger,
+                             io::HardwareI2c &i2c,
+                             const std::uint8_t channel)
+    : logger_(logger),
+      i2c_(i2c),
+      channel_(channel)
+{
+}
+
+Accelerometer::~Accelerometer()
+{
+}
+
+std::uint8_t Accelerometer::getChannel()
+{
+  return channel_;
+}
+
+std::optional<std::int16_t> Accelerometer::getRawAcceleration(const Axis axis)
+{
+  std::uint8_t low_byte_address, high_byte_address;
+  setRegisterAddressFromAxis(axis, &low_byte_address, &high_byte_address);
+  const auto low_byte = i2c_.readByte(kDeviceAddress, low_byte_address);
+  if (!low_byte) {
+    logger_.log(core::LogLevel::kFatal,
+                "Failed to read the low byte for acceleration along the %s",
+                AxisStrings[axis]);
+    return std::nullopt;
+  }
+  const auto high_byte = i2c_.readByte(kDeviceAddress, high_byte_address);
+  if (!high_byte) {
+    logger_.log(core::LogLevel::kFatal,
+                "Failed to read the high byte for acceleration along the %s",
+                AxisStrings[axis]);
+    return std::nullopt;
+  }
+  const std::int16_t raw_acceleration
+    = static_cast<std::int16_t>((high_byte.value() << 8) | low_byte.value());
+  return raw_acceleration;
+}
+
+void Accelerometer::setRegisterAddressFromAxis(const Axis axis,
+                                               std::uint8_t *low_byte_address,
+                                               std::uint8_t *high_byte_address)
+{
+  switch (axis) {
+    case Axis::x:
+      *low_byte_address  = kXOutLow;
+      *high_byte_address = kXOutHigh;
+      break;
+    case Axis::y:
+      *low_byte_address  = kYOutLow;
+      *high_byte_address = kYOutHigh;
+      break;
+    case Axis::z:
+      *low_byte_address  = kZOutLow;
+      *high_byte_address = kZOutHigh;
+      break;
+  }
+}
+
+std::int16_t Accelerometer::getAccelerationFromRaw(const std::int16_t rawAcceleration)
+{
+  // these values come from the data sheet. Don't change them.
+  std::int16_t acceleration
+    = static_cast<std::int16_t>((static_cast<int32_t>(rawAcceleration) * 488) / 1000);
```

Comment:
why are these casting to different sizes of int? it isn't definite that `int32_t*488/1000` will fit into an `int16_t`